controllers/controllers/clusters/vsphere.go

```diff
@@ -135,5 +137,49 @@ func (v *VSphereClusterReconciler) Reconcile(ctx context.Context, cluster *anywh
 		return reconciler.Result{}, err
 	}
 
+	workerNodeGroupMachineSpecs := make(map[string]anywherev1.VSphereMachineConfigSpec, len(machineConfigMap))
```

Comment:
should it be `len(machineConfigMap)-1` ? :0

Reply:
using len(cluster.Spec.WorkerNodeGroupConfigurations)